src/components/deployment/actions.ts

```diff
@@ -0,0 +1,122 @@
+import { Context } from 'koa';
+
+import { IAction, NotFound, ValidationError } from '../../lib';
+import * as entity from './entity';
+import { IDeployment, IEntity, IPaginatedEntities } from './types';
+import { validateDeployment } from './validation';
+
+function normalize(s: string, defaultValue: number): number {
+  const i = parseInt(s, 10) || defaultValue;
+  return i > 0 ? i : defaultValue;
+}
+
+export async function listDeployments(ctx: Context): Promise<IAction<IPaginatedEntities>> {
+  const current = normalize(ctx.query.page, 1);
+  const limit = normalize(ctx.query.limit, 10);
+  const query = {
+    page: current,
+    limit,
+    organization: ctx.query.organization,
+    space: ctx.query.space,
+  };
+
+  const [results, deployments] = await Promise.all([
+    entity.countDeployments(ctx.db, query),
+    entity.listDeployments(ctx.db, query),
+  ]);
+  const pages = Math.ceil(results / limit);
+
+  return {
```

Comment:
This return statement is a bit _yolo_

I would pull out the pagination into either the body of the returning function, or another function